File: exercism.io/kotlin/phone-number/src/main/kotlin/PhoneNumber.kt
class PhoneNumber(num: String) {
    val number = parse(num.filter(Char::isDigit))
    val areaCode = number.substring(0, 3)

    private fun parse(s: String) : String {
        return when {
            s.length == 10 -> s
            s.length == 11 && s[0] == '1' -> s.substring(1)
            else -> "0000000000"
        }
    }

    override fun toString() : String =
            String.format("(%s) %s-%s", areaCode, number.substring(3, 6), number.substring(6))
}

File: exercism.io/javascript/phone-number/phone-number.js
/**
 * Created by mewx on 19/12/16.
 */

var PhoneNumber = function (str) {
    this.phoneNumber = str.match(/\d/g).join("");
    if (this.phoneNumber.length < 10)
        this.phoneNumber = "0000000000";
    else if (this.phoneNumber.length == 11)
        this.phoneNumber = this.phoneNumber[0] == '1' ? this.phoneNumber.substring(1) : "0000000000";

    this.number = function () {
        return this.phoneNumber;
    };

    this.areaCode = function () {
        return this.phoneNumber.substring(0, 3);
    };

    this.toString = function () {
        return "(" + this.phoneNumber.substring(0, 3) + ") " +
            this.phoneNumber.substring(3, 6) + "-" + this.phoneNumber.substring(6);
    };
};

module.exports = PhoneNumber;
